Generator/Sources/NeedleFramework/Parsing/FileFilters/ComponentImplFilter.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Copyright (c) 2018. Uber Technologies
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+/// A filter that performs checks if the file content contains any
+/// component class implementations.
+class ComponentImplFilter: FileFilter {
```

Comment:
Seems like this is extremely similar to `ComponentExtensionFilter`: to the point where they could be the same class that takes a regex and a list of things to check (one of which would happen to have only one member).

Reply:
I didn't want to explode logic of the regex and component to name mapping to the task that uses the filters, but rather keep the filters themselves self-contained.

Reply:
You could subclass the base class in these files and pass the regex + name in their inits.

Reply:
I'll try the subclass approach. That should work well.